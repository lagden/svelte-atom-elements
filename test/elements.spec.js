/* globals describe, afterEach, beforeAll, afterAll, test, expect */

import timekeeper from 'timekeeper'
import {cleanup, render} from '@testing-library/svelte'
import {
	// Base
	Checkbox,
	Input,
	Label,
	Select,
	// Compose
	Check,
	CheckGroup,
	Combo,
	Datalist,
	Radio,
	Read,
	Request,
	Text,
} from '../src/elements.js'

import SlotLabel from './SlotLabel.svelte'

beforeAll(() => {
	// Para o tempo
	timekeeper.freeze(1_604_416_038 * 1000)
})

afterAll(() => {
	timekeeper.reset()
})

const options = [
	{value: 1, text: 'Lagden'},
	{value: 2, text: 'Takamoto'},
	{value: 3, text: 'Tadashi'},
	{value: 4, text: 'Rita'},
	{value: 5, text: 'Jorge'},
]

describe('Checkbox', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Checkbox)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Checkbox, {
			props: {
				checked: true,
				switchMode: true,
				class: 'customCss',
				required: true,
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Input', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Input)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Input, {
			props: {
				value: 'foo',
				class: 'customCss',
				type: 'tel',
				required: true,
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('invalid type', () => {
		const {container} = render(Input, {
			props: {
				type: 'xxx',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Label', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Label, {
			props: {
				for: 'inputID',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Label, {
			props: {
				class: 'customCss',
				for: 'inputID',
				required: true,
				disabled: false,
				'data-test': 1,
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot with slot', () => {
		const {container} = render(SlotLabel)
		expect(container).toMatchSnapshot()
	})
})

describe('Select', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Select)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Select, {
			props: {
				value: 2,
				options,
				first: 'Selecione',
				first_disabled: false,
				class: 'customCss',
				required: true,
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Check', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Check, {
			props: {
				name: 'test',
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Check, {
			props: {
				name: 'test',
				class: 'customCss',
				checked: true,
				showError: false,
				label: 'Do you agree with the terms?',
				labelCheck: 'agree',
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('CheckGroup', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(CheckGroup, {
			props: {
				name: 'test',
				id: 'test_choices',
				disabled: true,
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(CheckGroup, {
			props: {
				name: 'test',
				class: 'customCss',
				group: ['b', 'c'],
				options: [
					{value: 'a', text: 'a'},
					{value: 'b', text: 'b'},
					{value: 'c', text: 'c'},
				],
				showError: false,
				label: 'Choices:',
				id: 'test_choices',
				required: true,
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Combo', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Combo, {
			props: {
				name: 'test',
				outline: false,
				showError: false,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Combo, {
			props: {
				name: 'test',
				label: '<b>Test</b>',
				class: 'customCss',
				required: true,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Datalist', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Datalist, {
			props: {
				name: 'test',
				id: 'test',
				list: 'test_list',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Datalist, {
			props: {
				name: 'test',
				value: 'Rita',
				options,
				class: 'customCss',
				required: true,
				id: 'test',
				list: 'test_list',
				label: '<b>Test</b>',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Radio', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Radio, {
			props: {
				name: 'test',
				showError: false,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Radio, {
			props: {
				options,
				group: 1,
				name: 'test',
				toggleMode: true,
				class: 'customCss',
				label: '<b>Test</b>',
				required: true,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Read', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Read, {
			props: {
				value: 1,
				outline: false,
				wipe: true,
				showHelper: false,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Read, {
			props: {
				name: 'test',
				class: 'customCss',
				label: '<b>Test</b>',
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})

describe('Request', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Request, {
			props: {
				name: 'test',
				value: 1,
				outline: false,
				showError: false,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Request, {
			props: {
				name: 'test',
				class: 'customCss',
				label: '<b>Test</b>',
				type: 'tel',
				required: true,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})


describe('Text', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Text, {
			props: {
				name: 'test',
				value: 1,
				outline: false,
				showError: false,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Text, {
			props: {
				name: 'test',
				class: 'customCss',
				label: '<b>Test</b>',
				type: 'tel',
				required: true,
				id: 'test',
			},
		})
		expect(container).toMatchSnapshot()
	})
})
