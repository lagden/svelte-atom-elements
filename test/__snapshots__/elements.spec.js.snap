// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <label
        class="_atom_frm__label _atom_frm__label___checkbox"
        for="test"
      >
        <input
          class="_atom_frm__checkbox"
          id="test"
          name="test"
          type="checkbox"
        />
         
        
      </label>
       
      <small
        class="_atom_frm__error___message"
        title=""
      >
        
      </small>
    </div>
  </div>
</body>
`;

exports[`Check should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCss"
    >
      <span
        class="_atom_frm__label"
        id="test_label"
      >
        Do you agree with the terms?
      </span>
       
      <label
        class="_atom_frm__label _atom_frm__label___checkbox"
        for="test"
      >
        <input
          class="_atom_frm__checkbox"
          id="test"
          name="test"
          type="checkbox"
        />
         
        <span
          aria-label="agree"
        >
          agree
        </span>
        
      </label>
       
    </div>
  </div>
</body>
`;

exports[`CheckGroup should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <div
        aria-labelledby="test_choices_label"
        class="_atom_frm__group"
      />
    </div>
  </div>
</body>
`;

exports[`CheckGroup should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCss"
    >
      <span
        class="_atom_frm__label _atom_frm__label___required"
        id="test_choices_label"
      >
        Choices:
      </span>
       
      <div
        aria-labelledby="test_choices_label"
        class="_atom_frm__group"
      >
        
        <label
          class="_atom_frm__label _atom_frm__label___checkbox"
        >
          <input
            class="_atom_frm__checkbox"
            id="test_choices"
            name="test"
            required=""
            showerror="false"
            type="checkbox"
            value="a"
          />
           
          <span
            aria-label="a"
          >
            a
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___checkbox"
        >
          <input
            class="_atom_frm__checkbox"
            id="test_choices"
            name="test"
            required=""
            showerror="false"
            type="checkbox"
            value="b"
          />
           
          <span
            aria-label="b"
          >
            b
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___checkbox"
        >
          <input
            class="_atom_frm__checkbox"
            id="test_choices"
            name="test"
            required=""
            showerror="false"
            type="checkbox"
            value="c"
          />
           
          <span
            aria-label="c"
          >
            c
          </span>
           
        </label>
      </div>
    </div>
  </div>
</body>
`;

exports[`Checkbox should match snapshot default 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__checkbox"
      type="checkbox"
    />
  </div>
</body>
`;

exports[`Checkbox should match snapshot props 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__checkbox customCss _atom_frm__checkbox___switch"
      required=""
      type="checkbox"
    />
  </div>
</body>
`;

exports[`Combo should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <select
        class="_atom_frm__base _atom_frm__select _atom_frm__standard"
        id="test"
        name="test"
      >
        
      </select>
       
    </div>
  </div>
</body>
`;

exports[`Combo should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCss"
    >
      <label
        class="_atom_frm__label _atom_frm__label___required"
        for="test"
        required="true"
      >
        <b>
          Test
        </b>
        
      </label>
       
      <select
        class="_atom_frm__base _atom_frm__select _atom_frm__outline"
        id="test"
        name="test"
        required=""
      >
        
      </select>
       
      <small
        class="_atom_frm__error___message"
        title=""
      >
        
      </small>
    </div>
  </div>
</body>
`;

exports[`Input invalid type 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base "
      type="text"
    />
  </div>
</body>
`;

exports[`Input should match snapshot default 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base "
      type="text"
    />
  </div>
</body>
`;

exports[`Input should match snapshot props 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base customCss"
      required=""
      type="tel"
    />
  </div>
</body>
`;

exports[`Label should match snapshot default 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label"
    />
  </div>
</body>
`;

exports[`Label should match snapshot props 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label customCss _atom_frm__label___required"
      data-test="1"
      disabled="false"
      for="inputID"
      required="true"
    />
  </div>
</body>
`;

exports[`Label should match snapshot with slot 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label _atom_frm__label___disabled"
      disabled="true"
      for="test"
      required="false"
    >
      Test 
      <b>
        bold
      </b>
    </label>
  </div>
</body>
`;

exports[`Radio should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <div
        aria-labelledby="test_label"
        class="_atom_frm__group"
      />
       
    </div>
  </div>
</body>
`;

exports[`Radio should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCss"
    >
      <span
        class="_atom_frm__label _atom_frm__label___required"
        id="test_label"
      >
        <b>
          Test
        </b>
      </span>
       
      <div
        aria-labelledby="test_label"
        class="_atom_frm__group _atom_frm__group___toggle"
      >
        
        <label
          class="_atom_frm__label _atom_frm__label___radio"
          togglemode="true"
        >
          <input
            class="_atom_frm__radio _atom_frm__radio___toggle"
            id="test"
            name="test"
            required=""
            type="radio"
            value="1"
          />
           
          <span
            aria-label="Lagden"
          >
            Lagden
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___radio"
          togglemode="true"
        >
          <input
            class="_atom_frm__radio _atom_frm__radio___toggle"
            id="test"
            name="test"
            required=""
            type="radio"
            value="2"
          />
           
          <span
            aria-label="Takamoto"
          >
            Takamoto
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___radio"
          togglemode="true"
        >
          <input
            class="_atom_frm__radio _atom_frm__radio___toggle"
            id="test"
            name="test"
            required=""
            type="radio"
            value="3"
          />
           
          <span
            aria-label="Tadashi"
          >
            Tadashi
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___radio"
          togglemode="true"
        >
          <input
            class="_atom_frm__radio _atom_frm__radio___toggle"
            id="test"
            name="test"
            required=""
            type="radio"
            value="4"
          />
           
          <span
            aria-label="Rita"
          >
            Rita
          </span>
           
        </label>
        
        <label
          class="_atom_frm__label _atom_frm__label___radio"
          togglemode="true"
        >
          <input
            class="_atom_frm__radio _atom_frm__radio___toggle"
            id="test"
            name="test"
            required=""
            type="radio"
            value="5"
          />
           
          <span
            aria-label="Jorge"
          >
            Jorge
          </span>
           
        </label>
      </div>
       
      <small
        class="_atom_frm__error___message"
        title=""
      >
        
      </small>
    </div>
  </div>
</body>
`;

exports[`Select should match snapshot default 1`] = `
<body>
  <div>
    <select
      class="_atom_frm__base _atom_frm__select "
    >
      
    </select>
  </div>
</body>
`;

exports[`Select should match snapshot props 1`] = `
<body>
  <div>
    <select
      class="_atom_frm__base _atom_frm__select customCss"
      required=""
    >
      <option
        disabled=""
        value=""
      >
        Selecione
      </option>
      
      <option
        value="1"
      >
        Lagden
      </option>
      <option
        value="2"
      >
        Takamoto
      </option>
      <option
        value="3"
      >
        Tadashi
      </option>
      <option
        value="4"
      >
        Rita
      </option>
      <option
        value="5"
      >
        Jorge
      </option>
      
    </select>
  </div>
</body>
`;

exports[`Text should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <input
        class="_atom_frm__base _atom_frm__standard"
        id="test"
        name="test"
        type="text"
      />
       
    </div>
  </div>
</body>
`;

exports[`Text should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCss"
    >
      <label
        class="_atom_frm__label _atom_frm__label___required"
        for="test"
        required="true"
      >
        <b>
          Test
        </b>
        
      </label>
       
      <input
        class="_atom_frm__base _atom_frm__outline"
        id="test"
        name="test"
        required=""
        type="tel"
      />
       
      <small
        class="_atom_frm__error___message"
        title=""
      >
        
      </small>
    </div>
  </div>
</body>
`;
